Extension/src/LanguageServer/extension.ts

```diff
@@ -155,11 +156,85 @@ function sendActivationTelemetry(): void {
     telemetry.logLanguageServerEvent("Activate", activateEvent);
 }
 
+async function checkVsixCompatibility(): Promise<void> {
+    // Check to ensure the correct platform-specific VSIX was installed.
+    const vsixManifestPath: string = path.join(util.extensionPath, ".vsixmanifest");
+    // Skip the check if the file does not exist, such as when debugging cpptools.
+    if (await util.checkFileExists(vsixManifestPath)) {
```

Comment:
Can you early out instead?

Reply:
I would like to keep this as is.  I have changes to push after 1.9.1 to maintain a persistent value indicating whether or not to display the warning.  (The user can click an ignore button, if they intentionally want to use x86 on x64).  That persistent value is reset any time the scenario changes, so would need to be reset when these IFs are skipped.  That is cleaner to do when not doing an early exit.